### PR TITLE
feat: Use configured (const) OFF URLs in the UI

### PR DESCRIPTION
**Pull request by hangy**
### What
- Replace some hard-coded OFF/Robotoff URLs with ones that can be configured
- I intentionally left out some donation URLs, and so on, as they're not the same as API URLs
- Before this, it was impossible to test hunger-games with a custom Robotoff and OFF instance

### CODE REVIEW

File: src/hooks/useOptions.ts [hunk 1]
```diff
@@ -14,9 +15,7 @@ export const useOptions = (fileName: Files, lang: string) => {
 
   React.useEffect(() => {
     axios
-      .get(
-        `https://static.openfoodfacts.org/data/taxonomies/${fileName}.full.json`,
-      )
+      .get(`${OFF_URL}/data/taxonomies/${fileName}.full.json`)
```
Comment (alexfauquette) [inline src/hooks/useOptions.ts:18]: Is the removal of `static.` intentional?

File: src/off.ts [hunk 2]
```diff
@@ -163,29 +165,23 @@ const offService = {
       console.error("setIngedrient: Missing text");
     }
 
-    return axios.patch(
-      `https://world.openfoodfacts.org/api/v3/product/${code}`,
-      {
-        product: { [`ingredients_text${lang ? `_${lang}` : ""}`]: text },
-      },
-    );
+    return axios.patch(`${OFF_API_URL_V3}product/${code}`, {
```
Comment (alexfauquette) [inline src/off.ts:168]:
```suggestion
    return axios.patch(`${OFF_API_URL_V3}/product/${code}`, {
```

File: src/off.ts [hunk 3]
```diff
@@ -163,29 +165,23 @@ const offService = {
...
-      "https://world.openfoodfacts.org/api/v3/product/test",
-      {
-        fields: "ingredients",
-        lc: lang,
-        tags_lc: lang,
-        product: {
-          lang,
-          [`ingredients_text_${lang}`]: text,
-        },
+    return await axios.patch(`${OFF_API_URL_V3}product/test`, {
```
Comment (alexfauquette) [inline src/off.ts:176]:
```suggestion
    return await axios.patch(`${OFF_API_URL_V3}/product/test`, {
```

File: src/off.ts [hunk 4]
```diff
@@ -72,17 +74,17 @@ const offService = {
...
   getProductEditUrl(barcode) {
     const lang = getLang();
     return `https://world${
       lang === "en" ? "" : "-" + lang
-    }.openfoodfacts.org/cgi/product.pl?type=edit&code=${barcode}`;
+    }.${OFF_DOMAIN}/cgi/product.pl?type=edit&code=${barcode}`;
   },
   getLogoCropsByBarcodeUrl(barcode) {
-    return `https://hunger.openfoodfacts.org/logos/search?barcode=${barcode}`;
+    return `https://hunger.${OFF_DOMAIN}/logos/search?barcode=${barcode}`;
```
Comment (alexfauquette) [inline src/off.ts:87]:
```suggestion
    return `${URL_ORIGINE}/logos/search?barcode=${barcode}`;
```

File: src/pages/bug/index.tsx [hunk 5]
```diff
@@ -49,7 +49,7 @@ export default function BugPage() {
         onClick={() => {
           axios
             .post(
-              "https://world.openfoodfacts.org/api/v3/product/123456789",
+              `${OFF_API_URL_V3}product/123456789`,
```
Comment (alexfauquette) [inline src/pages/bug/index.tsx:52]:
```suggestion
              `${OFF_API_URL_V3}/product/123456789`,
```

Comment (alexfauquette) [reply src/pages/bug/index.tsx:52]: All the other reference of `OFF_API_URL_V3` have this same pattern. But with the following def I assume the `/` is missing

```js
export const OFF_API_URL_V3 = `${OFF_URL}/api/v3`;
```

File: src/robotoff.ts [hunk 6]
```diff
@@ -214,7 +214,9 @@ const robotoff = {
     }
 
     return axios.get(
-      `${ROBOTOFF_API_URL}/predict/nutrient?ocr_url=https://static.openfoodfacts.org/images/products/${productCodeForOcrUrl}/${imgid}.json`,
+      `${ROBOTOFF_API_URL}/predict/nutrient?ocr_url=${encodeURIComponent(
```
Comment (alexfauquette) [inline src/robotoff.ts:217]: Not sure about why you need to use `encodeURIComponent` for the URL string

File: src/hooks/useOptions.ts (hunk 1 above)
Comment (hangy) [reply src/hooks/useOptions.ts:18]: nope. good catch!

**Comment by teolemon** [comment]
Interesting PR. That should have the added benefit of making adapting Hunger Games to other flavors easier.

File: src/off.ts [hunk 7]
```diff
@@ -95,7 +98,7 @@ const offService = {
 
   getNutritionToFillUrl({ page, country, creator, category, code = false }) {
     if (code) {
-      return `https://world.openfoodfacts.org/api/v0/product/${code}.json?fields=code,states,lang,image_nutrition_url,product_name,nutriments,images,creator,countries`;
+      return `${OFF_API_URL}product/${code}.json?fields=code,states,lang,image_nutrition_url,product_name,nutriments,images,creator,countries`;
```
Comment (alexfauquette) [inline src/off.ts:101]:
```suggestion
      return `${OFF_API_URL}/product/${code}.json?fields=code,states,lang,image_nutrition_url,product_name,nutriments,images,creator,countries`;
```

File: src/pages/packaging/index.tsx [hunk 8]
```diff
@@ -199,7 +200,7 @@ const Page = () => {
             sx={{ width: 150 }}
             onClick={() => {
               axios.patch(
-                `https://world.openfoodfacts.org/api/v3/product/${product.code}`,
+                `${OFF_API_URL_V3}product/${product.code}`,
```
Comment (alexfauquette) [inline src/pages/packaging/index.tsx:203]:
```suggestion
                `${OFF_API_URL_V3}/product/${product.code}`,
```

**Review by alexfauquette — APPROVED** [review]
Just spotted two last missing `/` Otherwise it looks good

**Pull request merged**